scio-core/src/main/scala/com/spotify/scio/nio/ScioIO.scala

```diff
@@ -42,5 +42,37 @@ trait ScioIO[T] {
   def tap(read: ReadP): Tap[T]
 
   def id: String
+}
+
+object ScioIO {
+  // scalastyle:off structural.type
+  type ReadOnly[T, R] =
+    ScioIO[T] {
+      type ReadP = R
+      type WriteP = Nothing
+    }
+
+  type Aux[T, R, W] =
```

Comment:
Could you explain a bit, what is the purpose of this?

Reply:
I use this alias to properly type methods returning ScioIO, which make the compiler keep track of the existential types.